crates/levm/src/call_frame.rs

```diff
@@ -32,4 +32,24 @@ impl CallFrame {
     pub fn pc(&self) -> usize {
         self.pc
     }
+
+    pub fn jump(&mut self, jump_address: U256) {
+        if !self.valid_jump(jump_address) {
+            // probably should halt/panic
+            dbg!("Invalid jump");
+            return;
+        }
+        self.pc = jump_address.as_usize() + 1;
+    }
+
+    fn valid_jump(&self, offset: U256) -> bool {
+        // In the future this should be the Opcode::Invalid and halt
+        let opcode = self
+            .bytecode
+            .get(offset.as_usize())
+            .copied()
+            .map(Opcode::from)
+            .unwrap_or(Opcode::STOP);
+        opcode == Opcode::JUMPDEST
+    }
```

Comment:
We can use another auxiliary function called `get_opcode_at(&self, offset: U256) -> Opcode` or `opcode_at(&self, offset: U256) -> Opcode`. And then:


```suggestion
    fn valid_jump(&self, offset: U256) -> bool {
        // In the future this should be the Opcode::Invalid and halt
        self.opcode_at(offset).eq(Opcode::JUMPDEST)
    }
   
    fn opcode_at(&self, offset: U256) -> Opcode {
        self.bytecode
            .get(offset.as_usize())
            .copied()
            .map(Opcode::from)
            .unwrap_or(Opcode::STOP)
    }
```

If you want you can even get rid of `valid_jump` and just use `self.opcode_at(offset).eq(Opcode::JUMPDEST)
` in the `jump`'s function `if`.

Reply:
I like this approach and having the `valid_jump` function, looks more descriptive for what the `if` is checking, adding it